src/index.ts

```diff
@@ -1,19 +1,20 @@
 /* eslint-disable require-jsdoc */
-import got, { Response, Got, Options, CancelableRequest, HTTPError, CancelError } from 'got/dist/source';
+import got, { Response, Got, Options, CancelableRequest, HTTPError, CancelError, Method } from 'got/dist/source';
 import { AbortSignal } from 'abort-controller';
 import NodeCache from 'node-cache';
 
-type GotOptions = Pick<Options, 'method' | 'timeout' | 'decompress' | 'json' | 'retry' | 'headers' | 'form' | 'followRedirect'>;
+type GotOptions = Pick<Options, 'method' | 'timeout' | 'decompress' | 'json' | 'retry' | 'headers' | 'form' | 'followRedirect' | 'path'>;
 interface RequestOptions extends GotOptions{
   abortSignal?: AbortSignal;
+  isForm?: boolean;
 }
 
 export interface IHttpClient {
-  get<T>(path: string, options?: RequestOptions): Promise<T>;
-  post<T>(path: string, options?: RequestOptions): Promise<T>;
-  patch<T>(path: string, options?: RequestOptions): Promise<T>;
-  put<T>(path: string, options?: RequestOptions): Promise<T>;
-  delete(path: string, options?: RequestOptions): Promise<void>;
+  get<T>(path: string, options?: Omit<RequestOptions, 'json'>): Promise<T>;
+  post<T>(path: string, data?: Record<string, unknown>, options?: Omit<RequestOptions, 'json'>): Promise<T>;
+  patch<T>(path: string, data?: Record<string, unknown>, options?: Omit<RequestOptions, 'json'>): Promise<T>;
+  put<T>(path: string, data?: Record<string, unknown>, options?: Omit<RequestOptions, 'json'>): Promise<T>;
```

Comment:
I feel `Record<string, unknown>` is too limiting. as we discussed earlier, JSON fragments are a thing now, and even if you won't send a `number` or a `string`, surely you might want to be able to send an array?
Also, `Record<number, ...>` is a thing. Yeah, I know it'll become strings in the end so maybe it's not the best choice, but to prevent it when using TS would be a bit suboptimal IMO.

Reply:
I have seen a lot of `post` requests that actually don't send a payload `(data)`

Reply:
`Record<string, unknown> `  is the type prescribed by `Got`...  couldn't get away  with specifying `object`